Add Buy it button to product layout with custom icon support

roman-dvorak opened the pull request:
This PR adds a Buy it button to the product layout with the following features:

## Features
- Added support for `shop_url` and `shop_icon` in product front matter
- Button displays custom PNG image if `shop_icon` is provided (HTTP link to PNG)
- Falls back to standard Bulma button with shopping cart icon if no custom icon
- Button is right-aligned to match price and product code styling
- Opens links in new window (`target="_blank"`)

## Usage
Add to product front matter:
```yaml
shop_url: "https://example-eshop.com/product/123"
shop_icon: "https://example.com/custom-button.png"  # optional
```

## Changes
- Modified `_layouts/product.html` to include the new button functionality
- Button is conditionally displayed only when `shop_url` is defined
- Maintains existing layout and styling consistency

roman-dvorak committed:
Add buy-it button to product layout

roman-dvorak committed:
Add Buy it button to product layout with custom icon support

- Added shop_url and shop_icon support to product layout
- Button displays custom PNG image if shop_icon is provided
- Falls back to standard Bulma button with shopping cart icon
- Button is right-aligned to match price and product code
- Opens links in new window

roman-dvorak committed:
Add documentation for Buy it button feature

- Added shop_url and shop_icon parameters to product pages documentation
- Included examples for both custom PNG button and standard Bulma button
- Added detailed description of button behavior and positioning

chrisrhymes commented:
Hi, thanks for your pull request. I can definitely see the use for this feature. There are a few things I've noticed when I test this out.

<img width="1071" height="391" alt="Screenshot 2025-07-18 at 19 50 47" src="https://github.com/user-attachments/assets/46a7dec8-3a9e-49a8-83e7-c09bb91c2200" />


- The button could do with some margin so it's not so close to the text above it.
- The button doesn't seem to be right aligned as described in the product page documentation. Maybe Bulma's flexbox helpers could be used for alignment: https://bulma.io/documentation/helpers/flexbox-helpers/
- It would be great if people could change the default button text, so they could say 'Buy it on ebay' for example.
- For the image only button, I think it would be worth adding some text within the button for accessibility. This could potentially be done using the `is-sr-only` so it will be announced by screen readers. https://bulma.io/documentation/helpers/visibility-helpers/#other-visibility-helpers 
- In terms of using `target="_blank"` to open the link in a new window, it's encouraged not to open links in a new window where possible for accessibility reasons: https://www.w3.org/WAI/WCAG21/Techniques/general/G200

roman-dvorak committed:
Solve PR feedback: improve Buy it button accessibility and flexibility

roman-dvorak committed:
Support multiple buttons with customizable styles

roman-dvorak committed:
Add buttons example to product1

roman-dvorak commented:
Hi, thank you for your review and notes. I hope that I’ve fixed all the mentioned points.

In addition, I’ve made the solution more generic, so it’s now possible to define and place multiple buttons. It is solved in the same way as it’s already done for features.

Documentation page is updated.

I’ve also added one button to the example page (Product1) to demonstrate this approach.

It now looks like this, for example:
<img width="894" height="164" alt="image" src="https://github.com/user-attachments/assets/36b4a93d-4765-4cc8-9431-c1e305a56ea6" />

single button with custom image:
<img width="894" height="164" alt="image" src="https://github.com/user-attachments/assets/0d83075c-8a4e-41cb-bcf0-b8de611be657" />

chrisrhymes committed:
Update button icon to button image and icon_class to icon for consistency with other components

chrisrhymes commented:
Hi @roman-dvorak thank you for the updates. Its looking good. I have made a couple of small changes for consistency:

- The button.icon is now button.image
- The button.icon_class is now button.icon to be consistent with the features list icons
- The need for `fas` class has been removed to make it consistent with other icon usage in the theme
